### PR TITLE
`buildLcpInputs()`, `solveLcp()`, and `LcpInputs` struct

### PR DESCRIPTION
**Pull request by michguo**
Summary of changes per-review round:

Round 2 Review
- Move mX initialization code from `solveLcp()` to `buildLcpInputs()`.
- Store LCP inputs in a `LcpInputs` struct.
- Delete `test_LcpSolver.cpp` unit test, and simply rely on the `test_ConstraintSolver.cpp` unit test instead.
- Rebase #53 

Round 1 Review
- Separate `solveConstrainedGroup()` into `buildLcpInputs()` and `solveLcp()`.

### CODE REVIEW

File: unittests/unit/test_LCPSolver.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#include "dart/constraint/BoxedLcpConstraintSolver.hpp"
+#include "dart/simulation/World.hpp"
+
+#include "TestHelpers.hpp"
+
+using namespace dart;
+
+TEST(LCPSolver, SIMPLE)
+{
+  std::shared_ptr<simulation::World> world
```
Comment (keenon) [inline unittests/unit/test_LCPSolver.cpp:46]: Same deal on this test as the previous PR. You may actually just want to copy-paste the world setup from the other unit test into this one, once you get it working, so that there's something to have here.

It's fine to break it up into lots of unit tests in one file. You can leave this test as just the "it's empty and doesn't crash" edge case, but I think it's key to test the "it's _not_ empty and doesn't crash" too

File: dart/constraint/BoxedLcpConstraintSolver.hpp [hunk 2]
```diff
@@ -116,6 +116,15 @@ class BoxedLcpConstraintSolver : public ConstraintSolver
   std::vector<s_t*> solveConstrainedGroup(
       ConstrainedGroup& group, simulation::World* world) override;
 
+  /// Build the inputs to the LCP from the constraint group.
+  bool buildLcpInputs(ConstrainedGroup& group);
```
Comment (keenon) [inline dart/constraint/BoxedLcpConstraintSolver.hpp:120]: I think we should consider renaming this function. When I first read the name, I expected it to return a struct with all the LCP inputs (`A` matrix, `b` vector, `upperBound` vector, `lowerBound` vector, `fIndex` vector). But it returns a boolean. Maybe `shouldBuildLcpInputs()`? Or something...

Comment (keenon) [reply dart/constraint/BoxedLcpConstraintSolver.hpp:120]: Also, actually, this suggests the idea of a method that returns the LCP inputs, which is what I thought this was at first. `struct LcpInputs buildLcpInputs(ConstraintGroup& group)`, where we have

```
struct LcpInputs {
Eigen::MatrixXs A;
Eigen::VectorXs b;
Eigen::VectorXs upperBound;
Eigen::VectorXs lowerBound;
Eigen::VectorXi fIndex;
};
```

What do you think? The downside of this change is it's just more surface area for people to have to learn about in the APIs to get things to work. The alternative is to leave it as is, and hide the setup of the LCP inside of `solveLcp`, which has certain advantages for simplicity.

File: dart/constraint/BoxedLcpConstraintSolver.hpp [hunk 3]
```diff
@@ -116,6 +116,15 @@ class BoxedLcpConstraintSolver : public ConstraintSolver
...
+
+  /// Solve LCP from inputs.
```
Comment (keenon) [inline dart/constraint/BoxedLcpConstraintSolver.hpp:122]: I know I don't always write good comments, but this is one of those "this is a stop sign" comments. Maybe "Setup and solve an LCP to enforce the constraints on the ConstrainedGroup"?

**Review by keenon — COMMENTED** [review]
This one also looks broadly good. A few minor comments about naming and comments, and then the same drill about wanting the test to actually hit the main code path a bit.

File: dart/constraint/BoxedLcpConstraintSolver.cpp [hunk 4]
```diff
@@ -187,29 +187,20 @@ void BoxedLcpConstraintSolver::setCachedLCPSolution(Eigen::VectorXs X)
 }
 
 //==============================================================================
-std::vector<s_t*> BoxedLcpConstraintSolver::solveConstrainedGroup(
-    ConstrainedGroup& group, simulation::World* world)
+LcpInputs BoxedLcpConstraintSolver::buildLcpInputs(ConstrainedGroup& group)
```
Comment (keenon) [inline dart/constraint/BoxedLcpConstraintSolver.cpp:190]: This is lovely! Nice refactor.

**Review by keenon — APPROVED** [review]
